syntax = "proto3";
package logtail.models;

message SpanEvent {
    uint64 Timestamp = 1;
    bytes TraceID = 2;
    bytes SpanID = 3;
    bytes TraceState = 4;
    bytes ParentSpanID = 5;
    bytes Name = 6;
    enum SpanKind {
        UNSPECIFIED = 0;
        INTERVAL = 1;
        SERVER = 2;
        CLIENT = 3;
        PRODUCER = 4;
        CONSUMER = 5;
    }
    SpanKind Kind = 7;
    uint64 StartTime = 8;
    uint64 EndTime = 9;
    map<string, bytes> Tags = 10;
    message InnerEvent {
        uint64 Timestamp = 1;
        bytes Name = 2;
        map<string, bytes> Tags = 3;
    }
    repeated InnerEvent Events = 11;
    message SpanLink {
        bytes TraceID = 1;
        bytes SpanID = 2;
        bytes TraceState = 3;
        map<string, bytes> Tags = 4;
    }
    repeated SpanLink Links = 12;
    enum StatusCode {
        Unset = 0;
        Ok = 1;
        Error = 2;
    }
    StatusCode Status = 13;
    map<string, bytes> ScopeTags = 14;
}
